src/upnp_xml.cc

```diff
@@ -144,30 +150,24 @@ void UpnpXMLBuilder::renderObject(const std::shared_ptr<CdsObject>& obj, size_t
             quirks->restoreSamsungBookMarkedPosition(item, &result);
 
         auto meta = obj->getMetadata();
-        std::string upnp_class = obj->getClass();
 
         for (auto&& [key, val] : meta) {
-            if (key == MetadataHandler::getMetaFieldName(M_DESCRIPTION)) {
-                tmp = val;
-                if ((stringLimit > 0) && (tmp.length() > stringLimit)) {
-                    tmp = tmp.substr(0, getValidUTF8CutPosition(tmp, stringLimit - 3));
-                    tmp.append("...");
-                }
-                result.append_child(key.c_str()).append_child(pugi::node_pcdata).set_value(tmp.c_str());
-            } else if (key == MetadataHandler::getMetaFieldName(M_TRACKNUMBER)) {
-                if (upnp_class == UPNP_CLASS_MUSIC_TRACK) {
-                    result.append_child(key.c_str()).append_child(pugi::node_pcdata).set_value(val.c_str());
-                }
-            } else if (key != MetadataHandler::getMetaFieldName(M_TITLE)) {
-                addField(result, key, val);
-            }
+            // Trim metadata value as needed
+            auto str = trimString(val).c_str();
```

Comment:
The result of `trimString` is temporary, therefore `str` points to _nowhere_. `str` needs to be a std:string to store the result and `c_str` can only be called when using it in `set_value`.

Reply:
Good catch. I should probably review some older code of mine now because I'm fairly certain I've seen that pattern before.